app/models/service.rb

```diff
@@ -115,10 +115,12 @@ def cinstance
   scope :permitted_for, ->(user = nil) {
     next all unless user
 
+    next none if user.no_services_allowed?
+
     account = user.account
     account_services = (account.provider? ? account : account.provider_account).services
-    self.merge(
-      account_services.merge(user.forbidden_some_services? ? where(id: user.member_permission_service_ids) : {})
+    merge(
+      account_services.merge(user.selected_services_allowed? ? where(id: user.member_permission_service_ids) : {})
```

Comment:
This is the main fix.

The query relied on `forbidden_some_services?`, which it its turn calls `has_access_to_all_services?`, which checks:
```
!admin_sections.include?(:services)
```

So, it considers that if `services` permission is not present, then all services are enabled.

While this is partially true, it means that a newly created **member** user is not supposed to have **any** permissions (i.e.everything should be prohibited) - will have `has_access_to_all_services? == true`. Which results in the behavior reported in https://issues.redhat.com/browse/THREESCALE-6412

It is true that we determine whether all services are allowed by checking whether "services" permission is `nil` (happens when permission with section `:services` is not present). But I changed the behavior to only 

So, I changed the behavior, and now having a `nil` "services" permission (non-existing admin section `services`) means "all services are allowed" **ONLY IF** there is at least one enabled section that has permissions per-service (`:partners`, `:plans`, `:monitoring`)

It might break some previous functionality, so users who previously had some access will not have it anymore. But I think this is necessary, because it fixes a "security breach".

This probably was made so intentionally in the past, see https://github.com/3scale/system/commit/9c818157907c6c76d10a315fa572ff02e5ea327a

But I still think it's not right. Currently a member with **no** permissions cannot see the menus for the services in the UI, so it means they shouldn't be able to access the specific URLs (mentioned in the JIRA) either.